### PR TITLE
UI::Presenter

### PR DESCRIPTION
**Pull request by chubberlisk**
Created `UI::Presenter` as the in-between for the command line UI and the use cases.

Spied on and stubbed all dependencies. Not sure what is recommended for all the test doubles in `presenter_spec.rb`. They're only used in this file but they make the file kind of long. Other idea is to move them into individual files and place them in our `spec/test_doubles` directory. What is recommended?

### CODE REVIEW

File: spec/unit/user_interface/presenter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    before { ui_presenter.execute({}) }
+
+    it 'can ask the user for the first player' do
+      expect(command_line.ask_user_for_first_player_called?).to eq(true)
+    end
+
+    it 'can start a new game' do
+      expect(start_new_game.execute_called?).to eq(true)
+      expect(start_new_game.execute_arguments).to eq(
+        { first_player: :ask_user_for_first_player_response }
```
Comment (yusufsheiqh) [inline spec/unit/user_interface/presenter_spec.rb:119]: Are the curly braces necessary around a hash parameter?

Comment (chubberlisk) [reply spec/unit/user_interface/presenter_spec.rb]: No, I kept them for readability and consistency. In some tests such as:

```ruby
```

Comment (chubberlisk) [reply spec/unit/user_interface/presenter_spec.rb:119]: No they aren't. I kept them for readability and consistency. For some tests, the expectation on the arguments is an array such as this test:

https://github.com/chubberlisk/tic_tac_toe/blob/993717b145203f807ec1131a234081a1483617e5/spec/unit/user_interface/presenter_spec.rb#L169-L175

Do you think it's better to remove them?

Comment (chubberlisk) [reply spec/unit/user_interface/presenter_spec.rb:119]: I think I'll remove the braces as they're not necessary.

**Pull request merged**